File: seed/csharp-sdk/literal/src/SeedLiteral/Headers/HeadersClient.cs
using System.Text.Json;
using SeedLiteral;

#nullable enable

namespace SeedLiteral;

public class HeadersClient
{
    private RawClient _client;

    public HeadersClient(RawClient client)
    {
        _client = client;
    }

    public async Task<SendResponse> SendAsync(SendLiteralsInHeadersRequest request)
    {
        var _headers = new Dictionary<string, string>()
        {
            { "X-Endpoint-Version", request.EndpointVersion.ToString() },
            { "X-Async", request.Async.ToString() },
        };
        var response = await _client.MakeRequestAsync(
            new RawClient.ApiRequest
            {
                Method = HttpMethod.Post,
                Path = "/headers",
                Headers = _headers
            }
        );
        string responseBody = await response.Raw.Content.ReadAsStringAsync();
        if (response.StatusCode >= 200 && response.StatusCode < 400)
        {
            return JsonSerializer.Deserialize<SendResponse>(responseBody);
        }
        throw new Exception();
    }
}

File: seed/php-sdk/literal/src/Headers/HeadersClient.php
<?php

namespace Seed\Headers;

use Seed\Core\RawClient;
use Seed\Headers\Requests\SendLiteralsInHeadersRequest;
use Seed\Types\SendResponse;
use Seed\Core\JsonApiRequest;
use Seed\Core\HttpMethod;
use JsonException;
use Exception;
use Psr\Http\Client\ClientExceptionInterface;

class HeadersClient
{
    /**
     * @var RawClient $client
     */
    private RawClient $client;

    /**
     * @param RawClient $client
     */
    public function __construct(
        RawClient $client,
    ) {
        $this->client = $client;
    }

    /**
     * @param SendLiteralsInHeadersRequest $request
     * @param ?array{baseUrl?: string} $options
     * @returns SendResponse
     */
    public function send(SendLiteralsInHeadersRequest $request, ?array $options = null): SendResponse
    {
        $headers = [];
        $headers['X-Endpoint-Version'] = $request->endpointVersion;
        $headers['X-Async'] = $request->async;
        try {
            $response = $this->client->sendRequest(
                new JsonApiRequest(
                    baseUrl: $this->options['baseUrl'] ?? $this->client->options['baseUrl'] ?? '',
                    path: "headers",
                    method: HttpMethod::POST,
                    headers: $headers,
                    body: $request,
                ),
            );
            $statusCode = $response->getStatusCode();
            if ($statusCode >= 200 && $statusCode < 400) {
                $json = $response->getBody()->getContents();
                return SendResponse::fromJson($json);
            }
        } catch (JsonException $e) {
            throw new Exception("Failed to deserialize response", 0, $e);
        } catch (ClientExceptionInterface $e) {
            throw new Exception($e->getMessage());
        }
        throw new Exception("Error with status code " . $statusCode);
    }

}
